dds/DCPS/transport/framework/TransportSendBuffer.cpp

```diff
@@ -377,24 +377,34 @@ SingleSendBuffer::resend_fragments_i(const SequenceNumber& seq,
   if (fragments_.empty() || requested_frags.empty()) {
     return;
   }
-  const BufferMap& buffers = fragments_[seq];
-  const OPENDDS_VECTOR(SequenceRange) psr =
-    requested_frags.present_sequence_ranges();
-  SequenceNumber sent = SequenceNumber::ZERO();
-  for (size_t i = 0; i < psr.size(); ++i) {
-    BufferMap::const_iterator it = buffers.lower_bound(psr[i].first);
-    if (it == buffers.end()) {
-      return;
-    }
-    BufferMap::const_iterator it2 = buffers.lower_bound(psr[i].second);
-    while (true) {
-      if (sent < it->first) {
-        resend_one(it->second);
-        sent = it->first;
-      }
-      if (it == it2) {
-        break;
+  const FragmentMap::const_iterator fm_it = fragments_.find(seq);
+  if (fm_it == fragments_.end()) {
+    return;
+  }
+  const BufferMap& buffers = fm_it->second;
+  const OPENDDS_VECTOR(SequenceRange)& psr = requested_frags.present_sequence_ranges();
+
+  SequenceNumber frag_min;
+  size_t i = 0;
+  BufferMap::const_iterator it = buffers.lower_bound(psr.front().first);
+  BufferMap::const_iterator end = buffers.lower_bound(psr.back().second);
+  if (end != buffers.end()) {
+    ++end;
+  }
+
+  // Iterate over both containers simultaneously
+  while (i < psr.size() && it != end) {
+    if (psr[i].second < frag_min) {
```

Comment:
Does `frag_min` need to start at 0 or is it ok at 1?

Reply:
Spec says minimum fragment number is 1, and we want this to be the minimum fragment number of the current fragment buffer (it's inserted into map by the maximum fragment number), so we want it to start at 1.